iris-mpc-cpu/src/execution/hawk_main.rs

```diff
@@ -665,14 +667,16 @@ impl<'a> InMemoryStore for IrisLoader<'a> {
         right_code: &[u16],
         right_mask: &[u16],
     ) {
-        for (side, code, mask) in izip!(
+        for (store, pool, code, mask) in izip!(
             &mut self.irises,
+            self.iris_pools.clone(),
             [left_code, right_code],
             [left_mask, right_mask]
         ) {
             let iris = GaloisRingSharedIris::try_from_buffers(self.party_id, code, mask)
                 .expect("Wrong code or mask size");
-            side.insert(vector_id, iris);
+            let iris = pool.numa_realloc_blocking(iris).unwrap();
+            store.insert(vector_id, iris);
```

Comment:
Does it make sense to bring the store itself to the NUMA node of the pool that is going to use it ?

Reply:
Although not strictly necessary, it does make sense and this is planned.